Add title support for HTML body elements

daveverwer opened the pull request:
Again, more to start a discussion than anything. I'm sure this test is way too specific, but `<div>` elements should be able to accept `title` attributes.

JohnSundell commented on Tests/PlotTests/HTMLTests.swift:
```diff
@@ -602,6 +602,13 @@ final class HTMLTests: XCTestCase {
     }
 
+    func testDivWithTitleAttribute() {
+        let html = HTML(.body(.div("Element"), .title("Title")))
```
The expected output of this expression is one `div` and one `title` element as siblings. I'm guessing that you'd like to do this instead:
```swift
HTML(.body(.div("Element", .title("Title"))))
```
Since attributes are applied to the element that they appear within.

There is already an API for the `title` attribute, but it's currently constrained to `HTML.AbbreviationContext` (see [this file](https://github.com/JohnSundell/Plot/blob/master/Sources/Plot/API/HTMLAttributes.swift#L480)), so perhaps a solution would be to change that constraint to `HTML.BodyContext` instead? If we want the `title` attribute to be applicable to all `<body>` elements, that is.

daveverwer replied:
> I'm guessing that you'd like to do this instead:

😳 You're absolutely right... That's the problem with committing code in a test that doesn't compile... It's really hard to know whether that code is wrong because it's not written yet, or because it's just wrong 😂

daveverwer commented:
Alright, I made some progress on this.

Before I started, I checked just how global these attributes should be and it turns out they should be **truly global**. [See the spec for details](https://html.spec.whatwg.org/multipage/dom.html#global-attributes). At this point my thought was to add `title` globally, but it'd clash with the `title` *element* there and I don't know what you'd want to do about that in terms of naming.

So, I changed my approach. Even though it's valid, it's unlikely you'd want to add a title attribute to any element inside the `head` element. Adding the attribute to `BodyContext` seems like a reasonable compromise, and should avoid the clash of the `title`'s. 😂

I added the extension to `Node` and removed the `AbbreviationContext`. At [this point](https://github.com/JohnSundell/Plot/pull/47/commits/52eb70dce0db404923f6e9a90676dc7cd3076db9) everything is working and the tests pass. I did want to add one more check to my test though. I added an `a` tag and the compiler chokes on the `title` method for the `a` element.

I don't really know how to fix this. It doesn't seem right to re-implement `title` in the `LinkContext`  but without making `LinkContext` a `class`, I don't know how to fix it.

What do you think?

JohnSundell commented:
This looks really great @daveverwer, and totally agree with your reasoning. Let me check out your branch locally and give it a few spins, see if I can figure out what's going on with the `a` + `title` combination.

JohnSundell commented:
OK fixed it 😀 Converting this into a proper PR, and once CI passes, I'll merge it in 👍

daveverwer commented:
Fab! Thank you John.

Some more little tweaks have come up over the last few days, so there will be a couple more PRs in this vein on the way. Nothing outrageous though!

daveverwer commented on Sources/Plot/API/HTMLAttributes.swift:
```diff
@@ -103,9 +103,10 @@ public extension Node where Context == HTML.DocumentContext {
+
 public extension Node where Context == HTML.BodyContext {
     /// Specify a title for the element.
-    /// - parameter class: The title for the element.
```
Haha, who copies and pastes code?! NOT ME 😂

JohnSundell commented:
Sounds great, please keep 'em coming 😀
